sway-lsp/src/core/token_map.rs

```diff
@@ -104,7 +119,7 @@ impl TokenMap {
         tokens
             .filter_map(|(ident, token)| {
                 let span = match token.typed {
-                    Some(TypedAstToken::TypedFunctionDeclaration(decl)) => decl.span(),
+                    Some(TypedAstToken::TypedDeclaration(decl)) => decl.span(),
```

Comment:
This breaks autocompletion. The test should pass if we change this to:

```suggestion
                    Some(TypedAstToken::TypedDeclaration(decl)) => decl.span(),
                    Some(TypedAstToken::TypedFunctionDeclaration(decl)) => decl.span(),
```

Reply:
Thanks, i've changed this to

```rust
let span = match token.typed {
    Some(TypedAstToken::TypedFunctionDeclaration(decl)) if functions_only == Some(true) => decl.span(),
    Some(TypedAstToken::TypedDeclaration(decl)) => decl.span(),
    _ => ident.span(),
};
```